docs/fortranstyle.rst

```diff
@@ -305,6 +305,69 @@ Comments
       
       someStatementAfter
 
+
+Block constructs
+================
+
+* Block constructs are normally used in their verbose form, with block opening
+  and corresponding block closing in separate lines::
+
+    if (some_conditions) then
+      ! Do something
+      ...
+    end if
+
+* The closing form should have a space between the ``end`` keyword and the
+  construct type::
+
+    do ii = 1, 10
+      ...
+    end do  ! instead of "enddo"
+
+* If the block construct contains an expression within obligatory parentheses,
+  insert one space between the block type and the opening parenthesis::
+
+    if (some_condition) then  ! instead of "if(some_condition)"
+    
+    where (aa == 0)           ! instead of "where(aa == 0)"
+
+* Some block constructs have alternative one-line short forms without closing
+  statements (e.g. ``if``, ``where``). Only use their short form, if it is
+  readable and shorter than the verbose form. The short form should not exceed
+  two lines, and a line break (indicated by a continuation character pair) is
+  only allowed at the boundary between the construct header and the
+  expression/statement in its body. Line breaks within the header or within the
+  body are disallowed; if they become necessary, use the verbose form instead::
+
+    ! Allowed use cases of the short form
+    
+    if (allocated(someArray)) someArray(:,:) = 0.0_dp
+
+    where (aa /= 0.0_dp) aa = 1 / aa
+
+    if (some_complicated_exression .and. some_other_expression) &
+        & call someUsefulRoutine(aa, bb, cc, dd, ee, ff)
+
+    where (aa < sqrt(bb) - cc * dd) &
+        & aa = some_complicated_expression
```

Comment:
I would argue that even those statements are sometimes hard to read. My suggestion is to allow only a single line, which forces contributors to only apply this construct if it fits into the line. Of course, you can convince me if you do not see a problem here.

Reply:
Honestly, I don't have any convincing arguments, apart of that is something I think "I could eventually live with". :smile:  But I agree that this is already less readable as the one-line form. @bhourahine What is your opinion? If you also find the two-line form confusing, I am happy to drop it, and we then go with the one-line form only.

Reply:
@aradi I'd tend to agree with @terminationshock

Reply:
OK, I've changed the text accordingly, so that the single line form is allowed. (Makes also the description simpler...)